### ISSUE TITLE
[output] [spec] General

### ISSUE THREAD

**Issue by JeffSpies** [CenterForOpenScience/SHARE#131]
Based on @efc's spec, by default, we will only output events that have unique provider URLs--that is, any modification or deletion will not be represented in the system--only creations or modifications/deletions of records that we do not currently have data for.

- [ ] Confirmation that the `id/url` fields are unique representations of the record and not the event (creation, modification, deletion).
- [ ] Confirmation that the `id/url` fields should be used rather the `id/serviceID`. Some of the providers include multiple URLs; some providers also probably don't guarantee persistence of URLs in feeds.

Further, @efc's spec includes sorting by the date that scrapi collected the information--we are now calling that field `dateCollected`, which will be equivalent to `timestamps/consumeFinished`.

- [ ] Confirmation that `dateCollected` is the index to sort upon. 

I'd propose that we use `dateUpdated` as it is the date the provider uses to return the record for the date-based query we make and `dataCollected` is arbitrary. I'd also propose that we get rid of `dateCreated` as it almost has no consistency or substantive meaning, and we get rid of `dateCollected`.

**Comment by efc** [comment]
I think there are two possible "simple" routes we can take to making a consistent "event" record for the Notification Service. One (lets call this **Option A**), as @JeffSpies outlined above, is to simply say the first report we have of any given research release event is the only report we ever share as a "notification." However, since it appears to me that they Notification Service is really being designed to respond to queries rather than to really ship out real-time notifications, the other option (lets call this **Option B** would be to allow that "event" record to be updated to reflect any later changes reported by the provider.

Clearly Option A would be simpler to implement. However, the drawback is that we may get some providers who ship out preliminary "stub" records. There also may be providers who are diligent about fixing significant issues (wrong author, for example) and it would be a shame to not incorporate those changes. 

Option B would be moderately more difficult to implement, but still simple from the consumer perspective. In this case we would use subsequent retrievals of the same event (based on an ```id.url``` field match) to modify the "event" record for this resource. This would require maintaining an event record independent of the incoming raw record stream. Or, alternatively (may more like the model we have now), always making sure that the most recent "normalized" event record includes appropriate elements from past records merged into the new version and is the only copy of the normalized record actually indexed in elastic search.

Finally, whichever option we implement, we need to have a plan for take down requests, in case a provider made a serious error and now needs us to remove a record entirely. An alternative here might be the creation of a new-most-recent normalized record that says something like "withdrawn at provider request."

**Comment by efc** [comment]
As for sorting... 

I have already started using ```timestamps.consumeStarted``` as the field I sort by. I am not sure of the advantage of ```timestamps.consumeFinished``` but maybe someone can enlighten me. It seems the finished time is a bit more arbitrary since records sometimes finish consumption out of order (or at least that seemed to be the case in some brief reviews of this I did a couple weeks ago). I have no problem duplicating whichever of these is most appropriate to a ```dateCollected``` field.

Right now ```dateCreated``` and ```dateUpdated``` seem to be dates generated by the providers. The clearest meanings of these would be as reference to the original object: the date it was created and the date it was last updated. Of course, since these come from providers, we could make no guarantee that they have quite the same understanding of this field, though we could make some attempt to at least make these conform to whatever ISO format we standardize on in our normalized event records. I see the point @JeffSpies is making about recording the date used in the query, but I think that would be more clearly identified as a ```queryDate``` or a ```dateQueried```.
